### PR TITLE
Fix kill watch on macos / linux

### PR DESCRIPTION
**Pull request by TimWhiting**
Currently killing the watch command is not working on macos and linux

The following changes fix it. I've tested on both macos and linux

### CODE REVIEW

File: src/process.ts [hunk 1]
```diff
@@ -47,7 +52,7 @@ export class ChildProcessWrapper {
     if (isWin32) {
       return this.execSync(`taskkill /F /PID ${pid}`);
     } else {
-      return this.execSync(`kill -SIGINT ${pid}`);
+      return this.execSync(`kill -INT ${pid}`);
```
Comment (TimWhiting) [inline src/process.ts:55]: The child_process package uses the /bin/sh shell by default, where trying to use SIGINT fails. INT is the equivalent and works on both macos and linux.

File: src/process.ts [hunk 2]
```diff
@@ -28,17 +28,22 @@ export class ChildProcessWrapper {
...
         `(Get-WmiObject Win32_Process -Filter "ParentProcessID=${ppid}" | Where CommandLine -like '*build_runner watch*').ProcessId`,
         { encoding: "utf8", shell: "powershell.exe" }
       );
       console.log(parseInt(res));
       return res;
-    } else {
-      const res = this.execSync(`ps xao pid,ppid | grep "\d* ${ppid}"`);
-      return res.split(' ')[0];
+    } else if (isLinux) {
+      const res = this.execSync(`ps xao pid,ppid | grep -E "[[:digit:]]+[ ]+${ppid}"`);
```
Comment (TimWhiting) [inline src/process.ts:38]: In trying to fix this, I found out that grep regexes are very strange, this works on linux but not on macos, however as noted below macos does not show any child PIDs for the build_runner process.

File: src/process.ts (hunk 1 above)
Comment (gaetschwartz) [reply src/process.ts:55]: Thanks, lgtm !

File: src/process.ts (hunk 2 above)
Comment (gaetschwartz) [reply src/process.ts:38]: Are you sure bout it ? I'm pretty sure I remember being able to obtain a pid. Will look into that.

File: src/watch.ts [hunk 3]
```diff
@@ -110,7 +111,19 @@ export class BuildRunnerWatch {
...
+      //   \-+- child_pid1      <flutter_path>/bin/cache/dart-sdk/bin/dart --disable-dart-dev --packages=<flutter_path>/packages/flutter_tools/.packages <flutter_path>/bin/cache/flutter_tools.snapshot pub run build_runner watch
+      //    \--- child_pid2      <flutter_path>/bin/cache/dart-sdk/bin/dart __deprecated_pub run build_runner watch
+      // we thus need to kill child_pid2 by finding process.pid's grandchild
+      if (this.command === "flutter") {
+        const newPid = cp.getChildPID(pid);
+        if (newPid !== undefined) {
+          pid = newPid;
+        }
+      }
+
```
Comment (gaetschwartz) [inline src/watch.ts:126]: After some research it looks like when using flutter, it actually runs the dart command proxied so we need to go one depth deeper and find the grandchild's pid:
image
Whereas on dart: 
image

**Comment by TimWhiting** [comment]
Do you know when you will have time to finish this up & publish it? You made some comments, but also self-assigned and fixed some things yourself, so I don't know what to do about this. Just want to make sure you aren't waiting on me for anything.

**Comment by gaetschwartz** [comment]
Thanks for the reminder, I haven't had much time lately because of different life events, I will try to finalize this by this week :)

**Pull request merged**